content/en/blog/2022/apisix.md

```diff
@@ -0,0 +1,390 @@
+---
+title: Apache APISIX Integrates with OpenTelementry to Collect Tracing Data
+linkTitle: Apache APISIX-Opentelemetry Integration
+date: 2022-03-13
+author: Haochao Zhuang, Fei Han
+---
+
+This article introduces the Apache APISIX's `opentelemetry` plugin concept and
+how to enable and deploy the plugin.
+
+## Background Information
+
+OpenTelemetry is an open source telemetry data acquisition and processing
+system. It not only provides various SDKs for application side telemetry data
+collection and reporting, but also provides data collection side for data
+receiving, processing, and exporting. Export to any or more OpenTelemetry
+backends, such as Jaeger, Zipkin, and OpenCensus. You can view the list of
+plugins that have adapted the OpenTelemetry Collector in the [OpenTelemetry Registry](https://opentelemetry.io/registry/?s=collector&component=&language=).
+
+![Architecture-Present](/img/apisix/architecture-present.png)
+
+## Plugin Introduction
+
+The `opentelemetry` plugin of Apache APISIX implements Tracing data collection
+and sends it to OpenTelemetry Collector through HTTP protocol. Apache APISIX
+starts to support this feature in v2.13.0.
+
+One of OpenTelemetry's special features is that the Agent/SDK of OpenTelemetry
+is not locked with back-end implementation, which gives users flexibilities on
+choosing their own back-end services. In other words, users can choose the
+backend services they want, such as Zipkin and Jaeger, without affectting the
+application side.
+
+The `opentelemetry` plugin is located on the Agent side. It integrates the
+OpenTelemetry Agent/SDK and adopts its features in Apache APISIX. It can
+collect traced requests, generate `trace`, and forward them to the OpenTelemetry
+Collector. It supports the `trace` protocol, and it will support the `logs` and
+`metrics` protocols of OpenTelemetry in the next version.
+
+## Enable the Plugin
+
+You need to enable `opentelemetry` plugin and modify collector configuration in
+`conf/config.yaml` configuration file.
+
+We assume that you have already deployed the OpenTelemetry Collector on the same
+node as the APISIX and enabled the [OTLP HTTP Receiver](https://github.com/open-telemetry/opentelemetry-collector/blob/main/receiver/otlpreceiver/README.md).
+
+> If you have not completed the deployment, you can refer to the Scenario
+> Example section in the next section to complete the deployment of
+> OpenTelemetry Collector.
+
+The default port of the OTLP HTTP Receiver is `4318`, and the address of the
+`collector` is the HTTP Receiver address of the OpenTelemetry Collector. For
+related fields, please refer to the [Apache APISIX official documentation](https://apisix.apache.org/docs/apisix/next/plugins/opentelemetry/).
+
+For specific configuration, please refer to the following example:
+
+```shell
```

Comment:
This is YAML, is it not? If so, change to code-block language to reflect this.

Reply:
Yes, it is YAML. Updated.